R/param-estimates-batch.R

```diff
@@ -110,51 +110,117 @@ param_estimates_batch <- function(.path,
 #'   [stats::quantile()]. Represents the quantiles to calculate for parameter
-#'   estimates in `.param_df`.
+#'   estimates in `.boot_sum`.
 #' @param na.rm Logical scalar, passed through to [stats::quantile()].
 #'
 #' @importFrom tibble rownames_to_column
 #' @importFrom dplyr summarise across starts_with select rename left_join
 #' @importFrom stats quantile
+#'
+#' @return A tibble containing quantiles for each parameter estimate, optionally
+#' compared to the estimates from `.orig_mod`.
+#' @seealso summarize_bootstrap_run
+#' @examples
+#' \dontrun{
+#'
+#' # Via a bootstrap run
+#' boot_run <- read_model(file.path(MODEL_DIR, "1-boot"))
+#' boot_sum <- summarize_bootstrap_run(boot_run)
+#' param_estimates_compare(boot_sum)
+#'
+#' # Via a custom table
+#' orig_mod <- read_model(file.path(MODEL_DIR, "1"))
+#' param_df <- param_estimates_batch(MODEL_DIR)
+#' param_estimates_compare(param_df, .orig_mod = orig_mod)
+#' }
+#'
 #' @export
 param_estimates_compare <- function(
-  .param_df,
-  .orig_mod = NULL,
-  .compare_cols = starts_with(c("THETA", "SIGMA", "OMEGA")),
-  probs = c(.5, 0.025, 0.975),
-  na.rm = FALSE
-) {
-
-  comp_df <- .param_df %>%
+    .boot_sum,
+    .orig_mod = NULL,
+    .compare_cols = starts_with(c("THETA", "SIGMA", "OMEGA")),
+    probs = c(.5, 0.025, 0.975),
+    na.rm = FALSE
+){
+  UseMethod("param_estimates_compare")
+}
+
+#' @rdname param_estimates_compare
+#' @export
+param_estimates_compare.bbi_nmboot_summary <- function(
+    .boot_sum,
+    .orig_mod = NULL,
+    .compare_cols = NULL,
+    probs = c(.5, 0.025, 0.975),
+    na.rm = FALSE
+){
+
+  # Attempt to read in based_on model
+  if(is.null(.orig_mod)){
+    orig_mod_path <- fs::path_ext_remove(.boot_sum$based_on_model_path)
+    # make sure based_on model still exists
+    .orig_mod <- tryCatch(
+      read_model(orig_mod_path),
+      error = function(cond) NULL
+    )
+    if(!is.null(.orig_mod)){
+      orig_config_path <- file.path(
+        get_output_dir(.orig_mod, .check_exists = FALSE), "bbi_config.json"
+      )
+      if (fs::file_exists(orig_config_path)) {
+        res <- check_up_to_date(.orig_mod)
+        if(any(res[!res])){
+          rlang::warn("The original model is not up to date")
+        }
+      }else{
+        # Set the original model to NULL if the run directory/config file didnt
+        # exist
+        rlang::warn("The original model has not been run yet. Cannot compare estiamtes.")
+        .orig_mod <- NULL
+      }
+    }else{
+      rlang::warn(
+        c(
+          glue("The original model no longer exists at {orig_mod_path}"),
+          "Cannot compare to original model"
+        )
+      )
+    }
+  }
+
+  # Dont pass .compare_cols here, as we can only use columns in parameter_names,
+  # which could only be the default columns.
+  param_estimates_compare(
+    .boot_sum$boot_summary, .orig_mod = .orig_mod, probs = probs, na.rm = na.rm
+  )
```

Comment:
It would be nice to get more concise names in the output; possibly:

```r
> param_estimates_compare(boot_sum)
# A tibble: 12 × 5                                                                                                                                     
   parameter     original     p50   p2.5    p97.5
   <chr>            <dbl>   <dbl>   <dbl>   <dbl>
 1 THETA1           0.434   0.430   0.313   0.557 
 2 THETA2            4.12    4.12    4.06    4.17  
 3 THETA3            1.12    1.11    1.05    1.18  
 4 THETA4            4.21    4.21    4.18    4.25  
 5 THETA5            1.29    1.29    1.22    1.36  
 6 OMEGA(1,1)       0.221   0.215   0.117   0.329 
 7 OMEGA(2,1)      0.0690  0.0689  0.0286   0.110 
 8 OMEGA(2,2)      0.0827  0.0826  0.0631   0.103 
 9 OMEGA(3,1)       0.134   0.132  0.0881   0.181 
10 OMEGA(3,2)      0.0735  0.0731  0.0515  0.0933
11 OMEGA(3,3)       0.169   0.166   0.128   0.208 
12 SIGMA(1,1)      0.0399  0.0399  0.0375  0.0423
```

Currently has a large amount of white space between `parameter_names` and `original_estimate` due to long names:
```r
> param_estimates_compare(boot_sum)
# A tibble: 12 × 5                                                                                                                                     
   parameter_names original_estimate    p50   p2.5  p97.5
   <chr>                       <dbl>  <dbl>  <dbl>  <dbl>
 1 THETA1                     0.434  0.430  0.313  0.557 
 2 THETA2                     4.12   4.12   4.06   4.17  
 3 THETA3                     1.12   1.11   1.05   1.18  
 4 THETA4                     4.21   4.21   4.18   4.25  
 5 THETA5                     1.29   1.29   1.22   1.36  
 6 OMEGA(1,1)                 0.221  0.215  0.117  0.329 
 7 OMEGA(2,1)                 0.0690 0.0689 0.0286 0.110 
 8 OMEGA(2,2)                 0.0827 0.0826 0.0631 0.103 
 9 OMEGA(3,1)                 0.134  0.132  0.0881 0.181 
10 OMEGA(3,2)                 0.0735 0.0731 0.0515 0.0933
11 OMEGA(3,3)                 0.169  0.166  0.128  0.208 
12 SIGMA(1,1)                 0.0399 0.0399 0.0375 0.0423
```

Reply:
@seth127 I think im safe to change `original_estimate` --> `original`, but just wanted to make sure you're good with `parameter_names` --> `parameter` too (given how often we use that column name in other places). Given that this is solely an output, we definitely could, but just wanted to make sure consistency wasnt a concern here too.

Reply:
I don't have a problem with `original_estimate` --> `original`, but I want to keep `parameter_names` because, as @barrettk mentions, that's what we call that column in the output from a few other functions. 1) I'd like to keep this consistent and 2) this facilitates easier joining, when necessary.